### ISSUE TITLE
Only one of three defined cloudwatchevents created

### ISSUE THREAD

**Issue by pauldeden** [byu-oit-appdev/handel#79]
In my [handel.yml file](https://github.com/byu-oit-appdev/weekly-report-sla/blob/6761a57ba7b4b8da29451a9b2b1a061f1ebba784/handel.yml) three cloudwatchevents services are currently defined.

Only [one of them was created](https://us-west-2.console.aws.amazon.com/lambda/home?region=us-west-2#/functions/oit-weekly-reports-prd-function-lambda?tab=triggers) (the 2nd one) when my handel-codepipeline pipeline [ran in prd-appdev-oit-byu just now](https://us-west-2.console.aws.amazon.com/codepipeline/home?region=us-west-2#/view/oit_weekly_reports).

**Comment by dsw88** [comment]
It seems that event bindings, in particular in the consumeEvents phase, cannot be run in parallel. Since multiple producers are adding permissions to the consumer at the same time, AWS returns no permissions at first, and all three try to add a new permission document instead of just appending to an existing one. The result is that one of the calls wins and the others get lost.

We could try locking on just that part of the consume phase, but since that phase runs so quickly I'm going to change the events part of the lifecycle to run event bindings serially rather than in parallel, which should fix the issue.

**Comment by dsw88** [comment]
I serialized those two phases in the lifecycle, so things should work for multiple producers to the same consumer now. I didn't observe any noticeable slowdown in the overall deployment time as a result of serializing these phases because they are much faster than the deploy phase.

**Comment by pauldeden** [comment]
I just deleted the 4 cloudformation stacks for https://github.com/byu-oit-appdev/weekly-report-sla and reran the CodePipeline pipeline.  I had to manually delete the cloudwatch rules as the CloudFormation stacks errored out in deleting them because they still had targets (even after deleting the lambda function's cloudformation stack).

The CloudWatch rules created successfully.  Unfortunately, only one of the Lambda triggers created, this time the 3rd one.
